boa3_test/tests/examples_tests/test_ico.py

```diff
@@ -1,328 +1,515 @@
-        self.assertEqual(True, result)
+        total_supply_after = runner.call_contract(path, 'totalSupply')
+        owner_balance_after = runner.call_contract(path, 'balanceOf', owner_script_hash)
 
-        total_supply_after = self.run_smart_contract(engine, path, 'totalSupply')
-        self.assertEqual(total_supply_before + minted_amount, total_supply_after)
-        owner_balance_after = self.run_smart_contract(engine, path, 'balanceOf', self.OWNER_SCRIPT_HASH)
-        self.assertEqual(owner_balance_before + minted_amount, owner_balance_after)
+        runner.execute(account=self.OWNER)
+        self.assertEqual(VMState.HALT, runner.vm_state, msg=runner.error)
+
+        for x in range(len(invokes)):
+            self.assertEqual(expected_results[x], invokes[x].result)
+
+        self.assertEqual(total_supply_before.result + minted_amount, total_supply_after.result)
+        self.assertEqual(owner_balance_before.result + minted_amount, owner_balance_after.result)
 
     def test_ico_refund(self):
-        path = self.get_contract_path('ico.py')
-        engine = TestEngine()
-        engine.use_contract_custom_name = self._use_custom_name
-        transferred_amount = 10_000
+        path, _ = self.get_deploy_file_paths('ico.py')
+        runner = NeoTestRunner(runner_id=self.method_name())
+
+        invokes = []
+        expected_results = []
+
+        runner.add_gas(self.OWNER.address, self.GAS_TO_DEPLOY)
+        contract = runner.deploy_contract(path, account=self.OWNER)
+        runner.update_contracts(export_checkpoint=True)
+
+        transferred_neo_amount = 1
+        transferred_gas_amount = 10_000
+
+        owner_script_hash = self.OWNER.script_hash.to_array()
+        contract_script_hash = contract.script_hash
+
+        runner.add_neo(contract_script_hash, transferred_neo_amount * 2)
+        runner.add_gas(contract_script_hash, transferred_gas_amount * 2)
 
         # should fail script hash length is not 20
-        with self.assertRaisesRegex(TestExecutionException, self.ASSERT_RESULTED_FALSE_MSG):
-            self.run_smart_contract(engine, path, 'refund', bytes(10), transferred_amount, transferred_amount)
+        runner.call_contract(path, 'refund', bytes(10), transferred_neo_amount, transferred_gas_amount)
+        runner.execute()
+        self.assertEqual(VMState.FAULT, runner.vm_state, msg=runner.cli_log)
+        self.assertRegex(runner.error, self.ASSERT_RESULTED_FALSE_MSG)
 
         # should fail no amount is a positive number
-        with self.assertRaisesRegex(TestExecutionException, self.ASSERT_RESULTED_FALSE_MSG):
-            self.run_smart_contract(engine, path, 'refund', self.OWNER_SCRIPT_HASH, 0, 0)
+        runner.call_contract(path, 'refund', owner_script_hash, 0, 0)
+        runner.execute()
+        self.assertEqual(VMState.FAULT, runner.vm_state, msg=runner.cli_log)
+        self.assertRegex(runner.error, self.ASSERT_RESULTED_FALSE_MSG)
```

Comment:
Really minor adjustment, but since I already commented before, the VMState.FAULT tests should be at the bottom